### ISSUE TITLE
Over-enthusiastic removal of parentheses

### ISSUE THREAD

**Issue by SteveGilham** [fsprojects/fantomas#249]
Before
```
match i.Operand with
                            | :? (Instruction[]) -> true
                            | _ -> false
```
After
```
match i.Operand with
                            | :? Instruction [] -> true
                            | _ -> false
```
error FS0010: Unexpected symbol'[' in pattern matching. Expected '->' or other token.

**Comment by SteveGilham** [comment]
Also in cast expression
```
i.Operand :?> (Instruction [])
``
becoming 
```
i.Operand :?> Instruction []
```
with the same compiler error.

**Comment by jindraivanek** [comment]
First case fixed by #317.

Wasn't able to reproduce second case, for example `([] :> obj) :?> int seq` is valid code.

@SteveGilham, if you still see this as error, please post self-contained snippet for this case.
